pkg/importer/language.go

```diff
@@ -64,20 +68,103 @@ func (e *DiagnosticsError) String() string {
 
 func removeDuplicatePathedValues(pathedValues []PathedLiteralValue) []PathedLiteralValue {
 	uniqueValues := make([]PathedLiteralValue, 0)
-	occurrences := make(map[string]int)
+	identityOccurrences := make(map[string]int)
+	outputOccurrences := make(map[string]int)
 	for _, pathedValue := range pathedValues {
-		occurrences[pathedValue.Value]++
+		if pathedValue.Identity {
+			identityOccurrences[pathedValue.Value]++
+		} else {
+			outputOccurrences[pathedValue.Value]++
+		}
 	}
 
 	for _, pathedValue := range pathedValues {
-		if occurrences[pathedValue.Value] > 1 {
+		if pathedValue.Identity && identityOccurrences[pathedValue.Value] > 1 {
+			// an identity value that has occurred multiple times is not unique
+			continue
+		}
+
+		if !pathedValue.Identity && outputOccurrences[pathedValue.Value] > 1 {
+			// an output value that has occurred multiple times is not unique
 			continue
 		}
+
 		uniqueValues = append(uniqueValues, pathedValue)
 	}
 	return uniqueValues
 }
 
+func nextPropertyPath(path hcl.Traversal, key hcl.Traverser) hcl.Traversal {
+	newPath := make(hcl.Traversal, len(path))
+	copy(newPath, path)
+	newPath = append(newPath, key)
+	return newPath
```

Comment:
Isn't this just `return append(path, key)`? Why do we need a full copy?

Reply:
Because I thought `append` will mutate the path 🤔 lemme double check

Reply:
So it seems that `append` _does_ mutate in certain cases but not always. `return append(path, key)` will do the job here and unit tests are green. I will update it ✅